packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -634,6 +635,13 @@ function allocateInSlot(vm: VM, children: VNodes) {
             continue;
         }
 
+        // Dive one level further if the content is wrapped in a VFragment
+        if (isVFragment(vnode)) {
+            // Since the VFragment children are delimited by empty text nodes, we can ignore those
+            allocateInSlot(vm, vnode.children);
```

Comment:
I think the problem with this approach is that it will override any previously allocated children. Try this for example:
```xml
<template>
    <x-child>
          <div slot="slotname1">static content to named slot</div>
          <template lwc:if={condition}>
                 <span slot="slotname1">content allocated from if block</span>
          </template>
    </x-child>
</template>
```

Reply:
Thanks, that's a good callout. Will look into alternative solutions today.